### PR TITLE
Adding more Threadblock Tiles for Mixed-input TensorOp (BF16 * S8) in cutlass_library

### PR DESCRIPTION
**Pull request by manishucsd**
This PR adds more ThreadblockShapes (CTA) tile shapes in `cutlass_library`. More ThreadblockShapes are need to autotune and choose the most performant tile shapes for mixed-input GEMMs. 

## cutlass_library
For the `cutlass_library`, it adds more ThreadblockShapes into two `GenerateSM80*` function:
- `GenerateSM80_TensorOp_16816_mixed_input_upcast_a` : For upcast on operand A (e.g. S8* BF16)
- `GenerateSM80_TensorOp_16816_mixed_input_upcast_b` : For upcast on operand B (e.g. BF16 * S8)
- 

## Unit tests
More tile shapes needs different warp-configurations and this PR adds more device- and warp-level tests ensuring that those configurations are functionally solid (and remains solid).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # PyCache files
 __pycache__/
+cutlass_library.egg-info/
```
Comment (hwu36) [inline .gitignore:3]: what does this line do?

Comment (manishucsd) [reply .gitignore:3]: `cutlass_library.egg-info` is generated in the source folder and do we need it? I don't think we need this folder to be added to GitHub so I added to `.gitignore`. 

```bash
ls -l python/cutlass_library.egg-info/
total 16
-rw-rw-r-- 1 manigupta manigupta 167 Oct 10 18:41 PKG-INFO
-rw-rw-r-- 1 manigupta manigupta 550 Oct 10 18:41 SOURCES.txt
-rw-rw-r-- 1 manigupta manigupta   1 Oct 10 18:41 dependency_links.txt
-rw-rw-r-- 1 manigupta manigupta  16 Oct 10 18:41 top_level.txt
```

File: python/cutlass_library/generator.py [hunk 2]
```diff
@@ -2155,7 +2155,7 @@ def GenerateSM80_PlanarComplexTensorOp_16816(manifest, cuda_version):
 
 
 #
-def GenerateSM80_MixedInputTensorOp_16816(manifest, cuda_version):
+def GenerateSM80_TensorOp_16816_mixed_input_upcast_a(manifest, cuda_version):
```
[the remark this answers is not in the document]
Comment (manishucsd) [reply python/cutlass_library/generator.py:2158]: I didn't need to auto-tune `S8*BF16`. Thus, I didn't add tiles to upcast_a. 

We can add the tile shapes which are relevant for `S8*BF16` datatype for production GEMM shapes when the time comes?

File: test/unit/gemm/warp/gemm_mixed_input_sm80.cu [hunk 3]
```diff
@@ -120,6 +120,251 @@ TEST(SM80_warp_gemm_mixed_input_tensor_op_crosswise_i8_f16, 128x128x64_64x64x64_
 }
 
 
+TEST(SM80_warp_gemm_mixed_input_tensor_op_crosswise_i8_f16, 64x64x64_64x64x64_16x8x16) {
```
Comment (hwu36) [inline test/unit/gemm/warp/gemm_mixed_input_sm80.cu:123]: this test name repeats 8 times.

Comment (manishucsd) [reply test/unit/gemm/warp/gemm_mixed_input_sm80.cu]: Thanks for catching it. I fixed it. I was thinking how it got repeated, and it is probably during rebase against each of my local commits and me accepting the changes.

File: python/cutlass_library/generator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hwu36) [reply python/cutlass_library/generator.py:2158]: could we just copy paste `BF16*S8` ones here?  Just try to build and run them.  Remove those not runnable or buildible.

We can fine tune the stage number later.

[the remark this answers is not in the document]
Comment (manishucsd) [reply python/cutlass_library/generator.py:2158]: Added what worked for me. Can you please try this locally on your end?

```
make --no-warn-unused-cli -DCMAKE_BUILD_TYPE:STRING=Release -DCUTLASS_NVCC_ARCHS:STRING=80 -DCUTLASS_NVCC_KEEP:STRING=OFF -DCUTLASS_ENABLE_F16C:STRING=ON -DCUTLASS_DEBUG_TRACE_LEVEL:STRING=0 -DCUTLASS_LIBRARY_KERNELS:STRING=s16816gemm_s8_bf16_*,s16816gemm_bf16_s8_*,s16816gemm_bf16_128x128_64x3_tn_align8,

$ ./tools/profiler/cutlass_profiler --m=3456 --n=4096 --k=8192 --kernels=s16816gemm_s8_bf16_
```

**Review by hwu36 — APPROVED** [review]

**Pull request merged**